### PR TITLE
Convert SettingsPrivacyView to Tailwind

### PR DESCRIPTION
**Pull request by OceanOak**
No changelog

**Changes:**
-Reuse components to convert SettingsPrivacyView to Tailwind.
-Remove the top bar consent request.

**Before->After**
Frame 11 (1)

### CODE REVIEW

**Comment by OceanOak** [comment]
I believe we can delete _fullstory.scss file right?
What should we do about test `record_consent_saved_across_canvases` from the integration tests?
```
  test("record_consent_saved_across_canvases", async ({ page }, testInfo) => {
    await page.click("#fs-consent-yes");
    await page.waitForSelector(".fullstory-modal.hide");
    await page.waitForFunction(() => {
      let result = localStorage.getItem("userState-test");
      result = JSON.parse(result);
      return result["recordConsent"];
    });
    // navigate to another canvas
    await page.goto(canvasUrl("another-canvas"), {
      waitUntil: "networkidle",
    });

    await page.waitForSelector(".fullstory-modal.hide");

    // go back to original canvas to end the test
    const testname = testInfo.title;
    await page.goto(canvasUrl(testname));

    // there are errors loading some assets cause we're changing pages pretty fast.
    // Doesn't really matter, so ignore.
    clearMessages(testInfo);
  });
```

**Comment by StachuDotNet** [comment]
> What should we do about test `record_consent_saved_across_canvases` from the integration tests?

I haven't looked deeply into this, but could you update:

`await page.click("#fs-consent-yes");` to `await page.click("#toggle-fs-consent-true");`, or something like that?

**Comment by OceanOak** [comment]
> I haven't looked deeply into this, but could you update `await page.click("#fs-consent-yes");` to `await page.click("#toggle-fs-consent-true");` or something like that?

I couldn't run the integration tests to test this out. I get:
`Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?`

Trying to run `service docker status` or `service docker start` 
I got : `docker: unrecognized service`

the WSL 2 back-end for docker was previously enabled following [this guide](https://docs.docker.com/desktop/windows/wsl/)

Trying to run the integration tests locally also failed
`./integration-tests/run.sh: line 20: DARK_CONFIG_APISERVER_HOST: unbound variable`

**Comment by StachuDotNet** [comment]
> Trying to run the integration tests locally also failed
> `./integration-tests/run.sh: line 20: DARK_CONFIG_APISERVER_HOST: unbound variable`

Yeah, we have an issue right now where it's a bit annoying to run them locally. There's a little dance required to do such, noted at the top of `run.sh`. In addition to those instructions, locally comment out the 3 lines of that file that reference `PUBLISHED`. Can you let us know how that goes?

I think Paul may run the integration tests in the container, where you don't need this dance. Last time I tried, this failed, so I've been doing the above. Sorry for the headache! I've just filed an issue [here](https://github.com/darklang/dark/issues/4611) around this.

**Comment by pbiggar** [comment]
> await page.click("#fs-consent-yes");
> await page.waitForSelector(".fullstory-modal.hide");

I think these reference the modal you are removing? In that case, the test isn't useful and can be removed.

**Comment by OceanOak** [comment]
>  Can you let us know how that goes?

After some widows/wsl issues: `/bin/sh^M: bad interpreter: No such file or directory`  and `integration-tests/node_modules/.bin/playwright: 11: exec: node: not found`

23 test failed, 23 were skipped, and 24 passed.

This is the error message I got at the end:
```
Timed out waiting 600s for the entire test run
./integration-tests/run.sh: line 111: jq: command not found
```

**Comment by pbiggar** [comment]
Yeah, maybe running it in container will run better for you. It needs node, playwright, jq, bash, and other stuff to run outside it.

**Comment by pbiggar** [comment]
The integration tests are fine though - that's a known flaky test. Just rerun them until it's green.

File: client/src/components/Settings/SettingsPrivacyView.res [hunk 1]
```diff
@@ -7,86 +7,41 @@ module T = SettingsPrivacy
...
+        Html.span(list{Attrs.class(%twc("mt-6"))},list{toggle})
+        })
+      }
+    )}
   )
 }
 
 let view = (state: T.t): list<Html.html<AppTypes.msg>> => {
-  list{consentRow(state.recordConsent, ~longLabels=false)}
+  list{viewFSConsent(state.recordConsent|> Belt.Option.getExn)}
```
Comment (pbiggar) [inline client/src/components/Settings/SettingsPrivacyView.res:45]: We shouldn't be using getExn here. I guess we should make recordConsent a bool and default it to `true` - we don't use `None` anymore

File: client/src/app/AppTypes.res [hunk 2]
```diff
@@ -505,7 +505,7 @@ module SavedSettings = {
 
       {
         firstVisitToDark: oldFirstVisitToDark || newFirstVisitToDark,
-        recordConsent: withDefault(None, field("recordConsent", optional(bool)), j),
+        recordConsent: withDefault(default.recordConsent, field("recordConsent", bool), j),
```
Comment (pbiggar) [inline client/src/app/AppTypes.res:508]: This needs to still be able to read old values, so it needs to support reading both the old type and the new type. I think `optional(bool)` will safely read regular bools, but worth checking.

Should add a comment to go with it.

Comment (OceanOak) [reply client/src/app/AppTypes.res:508]: Isn't this why we're using withDefault or wouldn't it be enough alone? 
optional(bool) didn't work.

Comment (pbiggar) [reply client/src/app/AppTypes.res:508]: Why didnt `optional(bool)` work? Was there an error message?

The alternative is to try to parse it multiple times using different decoders. As I recall, there's a bs-json function that takes multiple decoders as an argument and tries them in turn.

Comment (OceanOak) [reply client/src/app/AppTypes.res:508]: - The error message is: 
This has type:  `Json_decode_extended.decoder<bool>`
Somewhere wanted: `Json_decode_extended.decoder<option<bool>>`

- Changing recordConsent type back to `option<bool>` will bring us back to where we started.

- There are two functions `oneOf` and `either`, both of them raise the same error.
`oneOf(list{optional(bool), field("recordConsent", bool)}, j),`
`either(optional(bool), field("recordConsent", bool), j),`

Is there something I'm missing or misunderstood?

Comment (pbiggar) [reply client/src/app/AppTypes.res:508]: `oneOf` is the right function. Both decoder arguments need to have the same type, so you need a decoder that returns a `bool`. Which means you need to write a decoder which handles the optional case and returns a default. There are example decoders is Json_decode_extended.

File: client/src/app/AppTypes.res [hunk 3]
```diff
@@ -505,7 +510,7 @@ module SavedSettings = {
 
       {
         firstVisitToDark: oldFirstVisitToDark || newFirstVisitToDark,
-        recordConsent: withDefault(default.recordConsent, field("recordConsent", bool), j),
+        recordConsent: oneOf(list{decodeopt, field("recordConsent", bool)}, j),
```
Comment (pbiggar) [inline client/src/app/AppTypes.res:513]: Something like:

```
recordConsent: field("recordConsent", oneOf(bool, (j) => optional(bool, j)->Option.unwrap(defaults.recordContent))
```

Comment (OceanOak) [reply client/src/app/AppTypes.res]: Using `Option.unwrap` I get
```
This has type: option<bool>
Somewhere wanted: Tc.Option.t<'a => 'b>
```
So, I've used `Belt.Option.getWithDefault` instead.

**Comment by pbiggar** [comment]
OK, looks good. Can you run the formatter before we merge?

**Comment by pbiggar** [comment]
Thanks so much!

**Pull request merged**